src/freenet/clients/http/wizardsteps/DATASTORE_SIZE.java

```diff
@@ -154,31 +151,38 @@ private long maxDatastoreSize() {
 		return slots * Node.sizePerKey;
 	}
 
-	private long canAutoconfigureDatastoreSize() {
-		if(!config.get("node").getOption("storeSize").isDefault())
-			return -1;
-
-		long freeSpace = core.node.getStoreDir().getUsableSpace();
-
-		if(freeSpace <= 0) {
-			return -1;
-		} else {
-			long shortSize;
-			if(freeSpace / 20 > 1024 * 1024 * 1024) { // 20GB+ => 5%, limit 256GB
-				// If 20GB+ free, 5% of available disk space.
-				// Maximum of 256GB. That's a 128MB bloom filter.
-				shortSize = Math.min(freeSpace / 20, 256*1024*1024*1024L);
-			}else if(freeSpace / 10 > 1024 * 1024 * 1024) { // 10GB+ => 10%
-				// If 10GB+ free, 10% of available disk space.
-				shortSize = freeSpace / 10;
-			}else if(freeSpace / 5 > 1024 * 1024 * 1024) { // 5GB+ => 512MB
-				// If 5GB+ free, default to 512MB
-				shortSize = 512*1024*1024;
-			}else { // <5GB => 256MB
-				shortSize = 256*1024*1024;
-			}
-
-			return shortSize;
-		}
-	}
+    private long canAutoconfigureDatastoreSize() {
+        if(!config.get("node").getOption("storeSize").isDefault())
+            return -1;
+
+        long freeSpace = core.node.getStoreDir().getUsableSpace();
+
+        if(freeSpace <= 0) {
+            return -1;
+        } else {
+            long shortSize;
+            long oneGiB = 1024 * 1024 * 1024L;
+            // Maximum for Freenet: 256GB. That's a 128MiB bloom filter.
+            long bloomFilter128MiBMax = 256 * oneGiB;
+            // Maximum to suggest to keep Disk I/O managable. This
+            // value might need revisiting when hardware or
+            // filesystems change.
+            long diskIoMax = 20 * oneGiB;
+            if(freeSpace / 50 > oneGiB) { // 50GB+ => 10%, but at least 10GiB
```

Comment:
May I suggest formatting these more like

    if (freeSpace > 50 * oneGiB) { // use 10% free space, or at least 10 GiB

I can do that if you like.